lib/rspec/mocks/proxy.rb

```diff
@@ -434,6 +440,11 @@ def warn(method_name)
         source = CallerFilter.first_non_rspec_line
         Kernel.warn("An expectation of :#{method_name} was set on nil. Called from #{source}. Use allow_message_expectations_on_nil to disable warnings.")
       end
+
+      def raise_error(method_name)
+        source = CallerFilter.first_non_rspec_line
+        raise("An expectation of :#{method_name} was set on nil. Called from #{source}. To allow expectations on nil, set disallow_expectations_on_nil to false.")
```

Comment:
I wonder if we have an appropriate exception type to either raise or subclass here rather than `StandardError`? I tend to like errors coming from the framework to be of a framework type, but can't remember what we do elsewhere in this project.